graphyte.py

```diff
@@ -64,31 +64,41 @@ def stop(self):
             self._thread.join()
             self.interval = None
 
-    def build_message(self, metric, value, timestamp):
+    def build_message(self, metric, value, timestamp, tags):
         """Build a Graphite message to send and return it as a byte string."""
         if not metric or metric.split(None, 1)[0] != metric:
             raise ValueError('"metric" must not have whitespace in it')
         if not isinstance(value, (int, float)):
             raise TypeError('"value" must be an int or a float, not a {}'.format(
-                    type(value).__name__))
-
-        message = u'{}{} {} {}\n'.format(
+                type(value).__name__))
+        if not (isinstance(tags, list) and all(isinstance(x, dict) for x in tags)):
+            raise TypeError('"tags" must be a list of dicts')
+        if not (all(all(isinstance(key, str) for key in tag.keys()) for tag in tags)):
+            raise TypeError('tag key must be an str')
```

Comment:
A couple of things here:

1) Why isn't the tags parameter simply a dict, rather than a list of dicts? Then you could just pass it as `tags={'tag1': 'value1', 'tag2': 'value2'}`. I don't see why the API should support multiple dicts of tags, and if the caller has that shape, they can merge before calling.
2) I realize we have a type check for the `value`, though that's a bit special. In general Python encourages "duck typing", meaning you don't check specific types but expect it to act like a duck (dict in this case). So I think we should just remove these type checks.